examples/experimental/bert_squad_pytorch/model_def.py

```diff
@@ -0,0 +1,166 @@
+        optimizer_grouped_parameters = [
+            {
+                "params": [
+                    p for n, p in model.named_parameters() if not any(nd in n for nd in no_decay)
+                ],
+                "weight_decay": self.context.get_hparam("weight_decay"),
+            },
+            {
+                "params": [
+                    p for n, p in model.named_parameters() if any(nd in n for nd in no_decay)
+                ],
+                "weight_decay": self.context.get_hparam("weight_decay"),
+            },
+        ]
+        optimizer = AdamW(
+            optimizer_grouped_parameters,
+            lr=self.context.get_hparam("learning_rate"),
+            eps=self.context.get_hparam("adam_epsilon")
+        )
+        return optimizer
+
+    def create_lr_scheduler(self, optimizer: torch.optim.Optimizer) -> LRScheduler:
+        scheduler = get_linear_schedule_with_warmup(
+            optimizer,
+            num_warmup_steps=self.context.get_hparam("num_warmup_steps"),
+            num_training_steps=self.context.get_hparam("num_training_steps"),
+        )
+        return LRScheduler(scheduler, LRScheduler.StepMode.STEP_EVERY_BATCH)
+
+    def train_batch(self, batch: TorchData, model: nn.Module, epoch_idx: int, batch_idx: int) -> Union[torch.Tensor, Dict[str, Any]] :
+        inputs = {
+            "input_ids": batch[0],
+            "attention_mask": batch[1],
+            "token_type_ids": batch[2],
+            "start_positions": batch[3],
+            "end_positions": batch[4],
+        }
+        outputs = model(**inputs)
+        loss = outputs[0]
+        self.tr_loss += loss.item()
+        return {"loss": self.tr_loss}
+
+    def evaluate_full_dataset(self, data_loader: DataLoader, model: nn.Module) -> Dict[str, Any]:
+        tokenizer = self.tokenizer_class.from_pretrained(
+            self.context.get_data_config().get("model_name_or_path"),
+            do_lower_case=True,
+            cache_dir=None
+        )
+
+        all_results = []
+        for batch in data_loader:
+            inputs = {
+                "input_ids": batch[0],
+                "attention_mask": batch[1],
+                "token_type_ids": batch[2],
+            }
+            feature_indices = batch[3]
+            outputs = model(**inputs)
+            for i, feature_index in enumerate(feature_indices):
+                eval_feature = self.features[feature_index.item()]
+                unique_id = int(eval_feature.unique_id)
+                output = [output[i].detach().cpu().tolist() for output in outputs]
+                start_logits, end_logits = output
+                result = SquadResult(unique_id, start_logits, end_logits)
+            all_results.append(result)
+
+        output_prediction_file = os.path.join(".", "predictions_{}.json".format(""))
+        output_nbest_file = os.path.join(".", "nbest_predictions_{}.json".format(""))
+        output_null_log_odds_file = None
```

Comment:
question: could we pass `None` instead of these filepaths to the transformers library `compute_predictions_logits` to prevent it from saving stuff to these paths? We're not using the result of these files in a Determined context, so better to avoid writing redundant stuff to disk.